### PR TITLE
New rule: Summary element has non-empty accessible name

### PR DESCRIPTION
**Pull request by WilcoFiers**
New rule to test that summary elements have an accessible name.

I don't think this is covered by any other rule today, since [summary elements do not have a semantic role](https://www.w3.org/TR/html-aria/#el-summary).

Need for Call for Review: 2 weeks

---

## How to Review And Approve

- Go to the “Files changed” tab
- Here you will have the option to leave comments on different lines.
- Once the review is completed, find the “Review changes” button in the top right, select “Approve” (if you are really confident in the rule) or "Request changes" and click “Submit review”.
- Make sure to also review the proposed Call for Review period. In case of disagreement, the longer period wins.

### CODE REVIEW

File: _rules/summary-non-empty-accessible-name-2t702h.md [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+---
+
+## Applicability
+
+This rule applies to HTML `summary` elements, all the following are true:
+
+- the element is [included in the accessibility tree][]; and
+- the element's [DOM tree][] parent is an HTML `details` element; and
+- the element is the first `summary` element of its siblings; and
+- the element has no [explicit role][], or an [explicit role][] which causes a [presentational role conflict][].
```
Comment (tombrunet) [inline _rules/summary-non-empty-accessible-name-2t702h.md]:
```suggestion
This rule applies to HTML `summary` elements where all of the following are true:

- the element is [included in the accessibility tree][]; and
- the element's [DOM tree][] parent is an HTML `details` element; and
- the element is the first `summary` element of its siblings; and
- the element has no [explicit role][], or the [explicit role][] is ignored due to a [presentational role conflict][].
```

File: _rules/summary-non-empty-accessible-name-2t702h.md [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+
+- The rule assumes that all `summary` elements are [user interface components as defined by WCAG 2](https://www.w3.org/TR/WCAG22/#dfn-user-interface-components).
+
+## Accessibility Support
+
+- Implementation of [Presentational Roles Conflict Resolution][] varies from one browser or assistive technology to another. Depending on this, some elements can have a [semantic role][] of `button` and fail this rule with some technology but users of other technologies would not experience any accessibility issue.
+
+## Background
+
+This rule is only applicable to `summary` elements that the browser will use as controls for a `summary` element. While this rule is not applicable to `summary` elements with an [explicit semantic role][], most of the time these likely do still require an [accessible name][]. This is covered by other rules, such as the [Button has non-empty accessible name][97a4e1].
```
Comment (tombrunet) [inline _rules/summary-non-empty-accessible-name-2t702h.md]:
```suggestion
This rule is only applicable to `summary` elements that the browser will use as controls for a `details` element. While this rule is not applicable to `summary` elements with an [explicit semantic role][], most of the time these likely do still require an [accessible name][]. This is covered by other rules, such as the [Button has non-empty accessible name][97a4e1].
```

**Review by tombrunet — CHANGES_REQUESTED** [review]
An inapplicable example for two summary elements to catch the third applicability bullet would be useful.

**Review by tombrunet — DISMISSED** [review]

**Review by kengdoj — DISMISSED** [review]

File: _rules/summary-non-empty-accessible-name-2t702h.md [hunk 3]
```diff
@@ -0,0 +1,193 @@
...
+
+## Accessibility Support
+
+- Implementation of [Presentational Roles Conflict Resolution][] varies from one browser or assistive technology to another. Depending on this, some elements can have a [semantic role][] of `button` and fail this rule with some technology but users of other technologies would not experience any accessibility issue.
+
+## Background
+
+This rule is only applicable to `summary` elements that the browser will use as controls for a `details` element. While this rule is not applicable to `summary` elements with an [explicit semantic role][], most of the time these likely do still require an [accessible name][]. This is covered by other rules, such as the [Button has non-empty accessible name][97a4e1].
+
+If the `summary` element is not included in the accessibility tree, but is still included in sequential focus navigation, this can result in accessibility issues not tested by this rule. This is covered under [Element with aria-hidden has no content in sequential focus navigation][6cfa84].
```
Comment (WilcoFiers) [inline _rules/summary-non-empty-accessible-name-2t702h.md:46]:
```suggestion
If the `summary` element is not included in the accessibility tree, but is still included in sequential focus navigation, this can result in accessibility issues not tested by this rule. This is covered under [Element with aria-hidden has no content in sequential focus navigation][6cfa84].

Note that some user agents expose the `summary` element with a `button` role. This deviates from the implicit ARIA semantics described in [HTML in ARIA](https://www.w3.org/TR/html-aria/#docconformance#el-summary). Because some browsers do not give `summary` elements a button role, these elements need to be tested separately from the [Button has non-empty accessible name](https://www.w3.org/WAI/standards-guidelines/act/rules/97a4e1/) ACT rule.
```

File: _rules/summary-non-empty-accessible-name-2t702h.md [hunk 4]
```diff
@@ -0,0 +1,195 @@
...
+
+Note that some user agents expose the `summary` element with a `button` role. This deviates from the implicit ARIA semantics described in [HTML in ARIA](https://www.w3.org/TR/html-aria/#docconformance#el-summary). Because some browsers do not give `summary` elements a button role, these elements need to be tested separately from the [Button has non-empty accessible name](https://www.w3.org/WAI/standards-guidelines/act/rules/97a4e1/) ACT rule.
```
Comment (Jym77) [inline _rules/summary-non-empty-accessible-name-2t702h.md]:
```suggestion
Note that some user agents expose the `summary` element with a `button` role. This deviates from the implicit ARIA semantics described in [HTML in ARIA](https://www.w3.org/TR/html-aria/#el-summary). Because some browsers do not give `summary` elements a button role, these elements need to be tested separately from the [Button has non-empty accessible name](https://www.w3.org/WAI/standards-guidelines/act/rules/97a4e1/) ACT rule.
```

File: _rules/summary-non-empty-accessible-name-2t702h.md [hunk 5]
```diff
@@ -0,0 +1,195 @@
...
+    - Wilco Fiers
+---
+
+## Applicability
+
+This rule applies to HTML `summary` elements where all of the following are true:
+
+- the element is [included in the accessibility tree][]; and
+- the element's [DOM tree][] parent is an HTML `details` element; and
+- the element is the first `summary` element of its siblings; and
```
Comment (Jym77) [inline _rules/summary-non-empty-accessible-name-2t702h.md]: Should it instead use "being a [summary for its parent details](https://html.spec.whatwg.org/multipage/interactive-elements.html#summary-for-its-parent-details)"?
Which is exactly the same, but using the HTML definition of it since it exists.

File: _rules/summary-non-empty-accessible-name-2t702h.md [hunk 6]
```diff
@@ -0,0 +1,195 @@
...
+</details>
+```
+
+#### Inapplicable Example 3
+
+This `summary` element has an explicit semantic role of `button`. These are tested under [Button has non-empty accessible name][97a4e1] instead.
+
+```html
+<details>
+	<summary role="button">Opening hours</summary>
```
Comment (Jym77) [inline _rules/summary-non-empty-accessible-name-2t702h.md:183]: As per the [ARIA in HTML definition of "No role"](https://www.w3.org/TR/html-aria/#dfn-no-role), I understand that it is an author error to use an explicit role on `summary`. Which is probably not what we want in Inapplicable example (notably, tools checking ARIA  conformance could fail that independently of the rule).

**Review by Jym77 — DISMISSED** [review]
I like it 👍

File: _rules/summary-non-empty-accessible-name-2t702h.md (hunk 6 above)
Comment (WilcoFiers) [reply _rules/summary-non-empty-accessible-name-2t702h.md:183]: I'm not sure I agree with that. Yes this is invalid, but WCAG doesn't require valid markup, and we often have inapplicable examples to demonstrate that invalid markup can still conform WCAG. If someone is failing this as a WCAG issue I think they made a mistake. I think an example that makes doing that an inconsistency with the rule is probably a good thing.

File: _rules/summary-non-empty-accessible-name-2t702h.md (hunk 5 above)
Comment (WilcoFiers) [inline _rules/summary-non-empty-accessible-name-2t702h.md]:
```suggestion
This rule applies to HTML `summary` elements [included in the accessibility tree][] that works as the [summary button for its parent `details` element][details summary].
```

File: _rules/summary-non-empty-accessible-name-2t702h.md [hunk 7]
```diff
@@ -0,0 +1,195 @@
...
+	<p>This is a website. We are available 24/7.</p>
+</details>
+```
+
+[accessible name]: #accessible-name 'Definition of accessible name'
+[explicit role]: #explicit-role 'Definition of explicit role'
+[focusable]: #focusable 'Definition of focusable'
+[included in the accessibility tree]: #included-in-the-accessibility-tree 'Definition of included in the accessibility tree'
+[presentational roles conflict resolution]: https://www.w3.org/TR/wai-aria-1.2/#conflict_resolution_presentation_none 'Presentational Roles Conflict Resolution'
+[semantic role]: #semantic-role 'Definition of Semantic Role'
```
Comment (WilcoFiers) [inline _rules/summary-non-empty-accessible-name-2t702h.md]:
```suggestion
[semantic role]: #semantic-role 'Definition of Semantic Role'
[details summary]: https://html.spec.whatwg.org/multipage/interactive-elements.html#summary-for-its-parent-details ' HTML 5 definition of summary for its parent details'
```

File: _rules/summary-non-empty-accessible-name-2t702h.md [hunk 8]
```diff
@@ -0,0 +1,192 @@
...
+	<div>
+		<summary></summary>
+	</div>
+	<p>This is a website. We are available 24/7.</p>
+</details>
+```
+
+#### Inapplicable Example 3
+
+This `summary` element has an explicit semantic role of `button`. These are tested under [Button has non-empty accessible name][97a4e1] instead.
```
Comment (WilcoFiers) [inline _rules/summary-non-empty-accessible-name-2t702h.md]:
```suggestion
This `summary` element has an explicit semantic role of `button`. These are tested under [Button has non-empty accessible name][97a4e1] instead. Note that while this example does not fail WCAG, [ARIA in HTML](https://www.w3.org/TR/html-aria/#dfn-no-role) does not allow overriding the role of a [summary for its parent details][details summary].
```

File: _rules/summary-non-empty-accessible-name-2t702h.md (hunk 6 above)
Comment (WilcoFiers) [reply _rules/summary-non-empty-accessible-name-2t702h.md:183]: As discussed. Added a note on this not being valid ARIA in HTML.

File: _rules/summary-non-empty-accessible-name-2t702h.md [hunk 9]
```diff
@@ -0,0 +1,192 @@
...
+  - CSS Styling
+acknowledgments:
+  authors:
+    - Wilco Fiers
+---
+
+## Applicability
+
+This rule applies to HTML `summary` elements [included in the accessibility tree][] that works as the [summary button for its parent `details` element][details summary].
+- the element has no [explicit role][], or the [explicit role][] is ignored due to a [presentational role conflict][].
```
Comment (WilcoFiers) [inline _rules/summary-non-empty-accessible-name-2t702h.md]:
```suggestion
This rule applies to HTML `summary` elements for which all the following is true:

- the element is  [included in the accessibility tree][]; and
- the element works as the [summary button for its parent `details` element][details summary]; and
- the element has no [explicit role][], or the [explicit role][] is ignored due to a [presentational role conflict].[].
```

File: _rules/summary-non-empty-accessible-name-2t702h.md [hunk 10]
```diff
@@ -0,0 +1,192 @@
...
+
+## Expectation
+
+Each target element has an [accessible name][] that is not empty (`""`).
+
+## Assumptions
+
+- The rule assumes that all `summary` elements are [user interface components as defined by WCAG 2](https://www.w3.org/TR/WCAG22/#dfn-user-interface-components).
```
Comment (Jym77) [inline _rules/summary-non-empty-accessible-name-2t702h.md]:
```suggestion
The rule assumes that all `summary` elements are [user interface components as defined by WCAG 2](https://www.w3.org/TR/WCAG22/#dfn-user-interface-components).
```

File: _rules/summary-non-empty-accessible-name-2t702h.md [hunk 11]
```diff
@@ -0,0 +1,192 @@
...
+
+## Accessibility Support
+
+- Implementation of [Presentational Roles Conflict Resolution][] varies from one browser or assistive technology to another. Depending on this, some elements can have a [semantic role][] of `button` and fail this rule with some technology but users of other technologies would not experience any accessibility issue.
```
Comment (Jym77) [inline _rules/summary-non-empty-accessible-name-2t702h.md]:
```suggestion
Implementation of [Presentational Roles Conflict Resolution][] varies from one browser or assistive technology to another. Depending on this, some elements can have a [semantic role][] of `button` and fail this rule with some technology but users of other technologies would not experience any accessibility issue.
```

File: _rules/summary-non-empty-accessible-name-2t702h.md [hunk 12]
```diff
@@ -0,0 +1,195 @@
...
+
+If the `summary` element is not included in the accessibility tree, but is still included in sequential focus navigation, this can result in accessibility issues not tested by this rule. This is covered under [Element with aria-hidden has no content in sequential focus navigation][6cfa84].
+
+Note that some user agents expose the `summary` element with a `button` role. This deviates from the implicit ARIA semantics described in [HTML in ARIA](https://www.w3.org/TR/html-aria/#el-summary). Because some browsers do not give `summary` elements a button role, these elements need to be tested separately from the [Button has non-empty accessible name](https://www.w3.org/WAI/standards-guidelines/act/rules/97a4e1/) ACT rule.
+
+### Bibliography
+
+- [HTML Accessibility API Mappings 1.0 (working draft), 5.2 `input type="button"`, `input type="submit"` and `input type="reset"`](https://www.w3.org/TR/html-aam/#input-type-button-input-type-submit-and-input-type-reset)
```
Comment (Jym77) [inline _rules/summary-non-empty-accessible-name-2t702h.md]: Is this relevant for this rule?

Comment (WilcoFiers) [inline _rules/summary-non-empty-accessible-name-2t702h.md]:
```suggestion
```

File: _rules/summary-non-empty-accessible-name-2t702h.md [hunk 13]
```diff
@@ -0,0 +1,194 @@
...
+
+## Background
+
+This rule is only applicable to `summary` elements that the browser will use as controls for a `details` element. While this rule is not applicable to `summary` elements with an [explicit semantic role][], most of the time these likely do still require an [accessible name][]. This is covered by other rules, such as the [Button has non-empty accessible name][97a4e1].
+
+If the `summary` element is not included in the accessibility tree, but is still included in sequential focus navigation, this can result in accessibility issues not tested by this rule. This is covered under [Element with aria-hidden has no content in sequential focus navigation][6cfa84].
+
+Note that some user agents expose the `summary` element with a `button` role. This deviates from the implicit ARIA semantics described in [HTML in ARIA](https://www.w3.org/TR/html-aria/#el-summary). Because some browsers do not give `summary` elements a button role, these elements need to be tested separately from the [Button has non-empty accessible name](https://www.w3.org/WAI/standards-guidelines/act/rules/97a4e1/) ACT rule.
```
Comment (giacomo-petri) [inline _rules/summary-non-empty-accessible-name-2t702h.md]:
```suggestion
Note that some user agents expose the `summary` element with a `button` role. This deviates from the implicit ARIA semantics described in [ARIA in HTML](https://www.w3.org/TR/html-aria/#el-summary). Because some browsers do not give `summary` elements a button role, these elements need to be tested separately from the [Button has non-empty accessible name](https://www.w3.org/WAI/standards-guidelines/act/rules/97a4e1/) ACT rule.
```

File: _rules/summary-non-empty-accessible-name-2t702h.md [hunk 14]
```diff
@@ -0,0 +1,194 @@
...
+
+### Bibliography
+
+- [Understanding Success Criterion 4.1.2: Name, Role, Value](https://www.w3.org/WAI/WCAG22/Understanding/name-role-value)
+- [ARIA14: Using aria-label to provide an invisible label where a visible label cannot be used](https://www.w3.org/WAI/WCAG22/Techniques/aria/ARIA14)
+- [ARIA16: Using aria-labelledby to provide a name for user interface controls](https://www.w3.org/WAI/WCAG22/Techniques/aria/ARIA16)
+
```
Comment (giacomo-petri) [inline _rules/summary-non-empty-accessible-name-2t702h.md:56]:
```suggestion
- [ARIA in HTML](https://www.w3.org/TR/html-aria/#el-summary)

```

File: _rules/summary-non-empty-accessible-name-2t702h.md [hunk 15]
```diff
@@ -0,0 +1,194 @@
...
+If the `summary` element is not included in the accessibility tree, but is still included in sequential focus navigation, this can result in accessibility issues not tested by this rule. This is covered under [Element with aria-hidden has no content in sequential focus navigation][6cfa84].
+
+Note that some user agents expose the `summary` element with a `button` role. This deviates from the implicit ARIA semantics described in [HTML in ARIA](https://www.w3.org/TR/html-aria/#el-summary). Because some browsers do not give `summary` elements a button role, these elements need to be tested separately from the [Button has non-empty accessible name](https://www.w3.org/WAI/standards-guidelines/act/rules/97a4e1/) ACT rule.
+
+### Bibliography
+
+- [Understanding Success Criterion 4.1.2: Name, Role, Value](https://www.w3.org/WAI/WCAG22/Understanding/name-role-value)
+- [ARIA14: Using aria-label to provide an invisible label where a visible label cannot be used](https://www.w3.org/WAI/WCAG22/Techniques/aria/ARIA14)
+- [ARIA16: Using aria-labelledby to provide a name for user interface controls](https://www.w3.org/WAI/WCAG22/Techniques/aria/ARIA16)
```
Comment (WilcoFiers) [inline _rules/summary-non-empty-accessible-name-2t702h.md:54]:
```suggestion
- [ARIA16: Using aria-labelledby to provide a name for user interface controls](https://www.w3.org/WAI/WCAG22/Techniques/aria/ARIA16)
- [Summary element entry for ARIA in HTML](https://www.w3.org/TR/html-aria/#el-summary)
```

File: _rules/summary-non-empty-accessible-name-2t702h.md [hunk 16]
```diff
@@ -0,0 +1,194 @@
...
+    - Wilco Fiers
+---
+
+## Applicability
+
+This rule applies to HTML `summary` elements for which all the following is true:
+
+- the element is  [included in the accessibility tree][]; and
+- the element works as the [summary button for its parent `details` element][details summary]; and
+- the element has no [explicit role][], or the [explicit role][] is ignored due to a [presentational role conflict].[].
```
Comment (Jym77) [inline _rules/summary-non-empty-accessible-name-2t702h.md]:
```suggestion
- the element has no [explicit role][], or the [explicit role][] is ignored due to a [presentational role conflict][].
```

File: _rules/summary-non-empty-accessible-name-2t702h.md [hunk 17]
```diff
@@ -0,0 +1,194 @@
...
+	<div>
+		<summary></summary>
+	</div>
+	<p>This is a website. We are available 24/7.</p>
+</details>
+```
+
+#### Inapplicable Example 3
+
+This `summary` element has an explicit semantic role of `button`. These are tested under [Button has non-empty accessible name][97a4e1] instead. Note that while this example does not fail WCAG, [ARIA in HTML](https://www.w3.org/TR/html-aria/#dfn-no-role) does not allow overriding the role of a [summary for its parent details][details summary].
```
Comment (Jym77) [inline _rules/summary-non-empty-accessible-name-2t702h.md]:
```suggestion
This `summary` element has an explicit semantic role of `button`. These are tested under [Button has non-empty accessible name][97a4e1] instead. Note that while this example does not fail WCAG, under [ARIA in HTML](https://www.w3.org/TR/html-aria/#dfn-no-role) it is not allowed to override the role of a [summary for its parent details][details summary].
```

**Review by Jym77 — DISMISSED** [review]
Typos.

**Review by Jym77 — APPROVED** [review]

File: _rules/summary-non-empty-accessible-name-2t702h.md [hunk 18]
```diff
@@ -0,0 +1,195 @@
...
+acknowledgments:
+  authors:
+    - Wilco Fiers
+---
+
+## Applicability
+
+This rule applies to HTML `summary` elements for which all the following is true:
+
+- the element is  [included in the accessibility tree][]; and
```
Comment (giacomo-petri) [inline _rules/summary-non-empty-accessible-name-2t702h.md]: removed double-space 

```suggestion
- the element is [included in the accessibility tree][]; and
```

**Review by giacomo-petri — APPROVED** [review]

**Review by Jym77 — APPROVED** [review]

File: _rules/summary-non-empty-accessible-name-2t702h.md [hunk 19]
```diff
@@ -0,0 +1,195 @@
...
+
+Each target element has an [accessible name][] that is not empty (`""`).
+
+## Assumptions
+
+The rule assumes that all `summary` elements are [user interface components as defined by WCAG 2](https://www.w3.org/TR/WCAG22/#dfn-user-interface-components).
+
+## Accessibility Support
+
+Implementation of [Presentational Roles Conflict Resolution][] varies from one browser or assistive technology to another. Depending on this, some elements can have a [semantic role][] of `button` and fail this rule with some technology but users of other technologies would not experience any accessibility issue.
```
Comment (carlosapaduarte) [inline _rules/summary-non-empty-accessible-name-2t702h.md]: Why would the element having a semantic role of button make it fail this rule? Is this copy/paste from another rule?

**Comment by giacomo-petri** [comment]
In Firefox, the ::marker CSS pseudo-element is part of the accessibility tree, which means the accessible name cannot be left empty in Firefox. 

Should we address this, clarifying that the "triangle" icon is not enough?

File: _rules/summary-non-empty-accessible-name-2t702h.md [hunk 20]
```diff
@@ -0,0 +1,195 @@
...
+
+This rule applies to HTML `summary` elements for which all the following is true:
+
+- the element is [included in the accessibility tree][]; and
+- the element works as the [summary button for its parent `details` element][details summary]; and
+- the element has no [explicit role][], or the [explicit role][] is ignored due to a [presentational role conflict][].
+
+## Expectation
+
+Each target element has an [accessible name][] that is not empty (`""`).
```
Comment (WilcoFiers) [inline _rules/summary-non-empty-accessible-name-2t702h.md]:
```suggestion
Each target element has an [accessible name][] that is not empty (`""`), nor just the name of the marker pseudo element.
```

File: _rules/summary-non-empty-accessible-name-2t702h.md [hunk 21]
```diff
@@ -0,0 +1,195 @@
...
+
+This rule applies to HTML `summary` elements for which all the following is true:
+
+- the element is [included in the accessibility tree][]; and
+- the element works as the [summary button for its parent `details` element][details summary]; and
+- the element has no [explicit role][], or the [explicit role][] is ignored due to a [presentational role conflict][].
+
+## Expectation
+
+Each target element has an [accessible name][] that is not empty (`""`), nor just the name of the marker pseudo element.
```
Comment (Jym77) [inline _rules/summary-non-empty-accessible-name-2t702h.md]:
```suggestion
Each target element has an [accessible name][] that is not empty (`""`), nor just the name of the `::marker` pseudo element.
```

File: _rules/summary-non-empty-accessible-name-2t702h.md [hunk 22]
```diff
@@ -0,0 +1,195 @@
...
+
+```html
+<details>
+	<summary>Opening times</summary>
+	<summary></summary>
+	<p>This is a website. We are available 24/7.</p>
+</details>
+```
+
+### Failed
```
Comment (carlosapaduarte) [inline _rules/summary-non-empty-accessible-name-2t702h.md:118]: Add failed example with two summary elements, being the first empty

File: _rules/summary-non-empty-accessible-name-2t702h.md [hunk 23]
```diff
@@ -0,0 +1,195 @@
...
+
+Each target element has an [accessible name][] that is not empty (`""`), nor just the name of the marker pseudo element.
+
+## Assumptions
+
+The rule assumes that all `summary` elements are [user interface components as defined by WCAG 2](https://www.w3.org/TR/WCAG22/#dfn-user-interface-components).
+
+## Accessibility Support
+
+Implementation of [Presentational Roles Conflict Resolution][] varies from one browser or assistive technology to another. Depending on this, some elements can have a [semantic role][] of `button` and fail this rule with some technology but users of other technologies would not experience any accessibility issue.
```
Comment (WilcoFiers) [inline _rules/summary-non-empty-accessible-name-2t702h.md]:
```suggestion
There is a difference in how user agents expose the triangle indicating the control's expand state. As a result, some user agents include the triangle in the accessible name of the summary element.
```

File: _rules/summary-non-empty-accessible-name-2t702h.md [hunk 24]
```diff
@@ -0,0 +1,195 @@
...
+
+This `summary` element has an [explicit role][] of `none`. However, it is [focusable][] (by default) which causes [Presentational Roles Conflict Resolution][]. It fails because it has an empty [accessible name][].
+
+```html
+<details>
+	<summary role="none"></summary>
+	<p>This is a website. We are available 24/7.</p>
+</details>
+```
+
```
Comment (WilcoFiers) [inline _rules/summary-non-empty-accessible-name-2t702h.md:141]:
````suggestion

#### Failed Example 3

This first `summary` element has no [accessible name][] because it is empty. The second `summary` element is inapplicable because only the first `summary` element will be used as a control for the `details` element.

```html
<details>
	<summary></summary>
	<summary>Opening times</summary>
	<p>This is a website. We are available 24/7.</p>
</details>
```

````

File: _rules/summary-non-empty-accessible-name-2t702h.md [hunk 25]
```diff
@@ -0,0 +1,195 @@
...
+	<summary></summary>
+	<p>This is a website. We are available 24/7.</p>
+</details>
+```
+
+### Failed
+
+#### Failed Example 1
+
+This `summary` element has no [accessible name][] because it has no content or attribute that can provide it.
```
Comment (WilcoFiers) [inline _rules/summary-non-empty-accessible-name-2t702h.md]:
```suggestion
This `summary` element has no [accessible name][], or an accessible name with just the `::marker` pseudo element, because it has no content or attribute that can provide it.
```

**Review by carlosapaduarte — APPROVED** [review]

File: _rules/summary-non-empty-accessible-name-2t702h.md [hunk 26]
```diff
@@ -0,0 +1,207 @@
...
+	<p>This is a website. We are available 24/7.</p>
+</details>
+```
+
+[accessible name]: #accessible-name 'Definition of accessible name'
+[explicit role]: #explicit-role 'Definition of explicit role'
+[focusable]: #focusable 'Definition of focusable'
+[included in the accessibility tree]: #included-in-the-accessibility-tree 'Definition of included in the accessibility tree'
+[presentational roles conflict resolution]: https://www.w3.org/TR/wai-aria-1.2/#conflict_resolution_presentation_none 'Presentational Roles Conflict Resolution'
+[semantic role]: #semantic-role 'Definition of Semantic Role'
```
Comment (WilcoFiers) [inline _rules/summary-non-empty-accessible-name-2t702h.md]:
```suggestion
```

**Review by colabottles — APPROVED** [review]

**Review by kengdoj — APPROVED** [review]

**Review by HelenBurge — APPROVED** [review]
